bundles/http/Citrix/Cherwell/Cherwell.js

```
@@ -0,0 +1,1278 @@
+    let moreRecords = true;
+    if (resp.ok && resp.status === 200) {
+        let data = await resp.json();
+        let dataArray = [];
+        let types = [];
+        for (let i = 0, j = 0; i < data.records.length; i++) {
+            dataArray[i] = JSON.parse(data.records[i]);
+            dataArray[i].CreatedDateTime = new Date(dataArray[i].CreatedDateTime);
+            dataArray[i].LastModifiedDateTime = new Date(dataArray[i].LastModifiedDateTime);
+            dataArray[i].LastModDateTime = new Date(dataArray[i].LastModDateTime);
+            dataArray[i].SLAResponseWarning = new Date(dataArray[i].SLAResponseWarning);
+            dataArray[i].SLAResolutionWarning = new Date(dataArray[i].SLAResolutionWarning);
+
+            if (moduleName == 'servicesIncidentTypes' || moduleName == 'categoriesIncidentTypes') {
+                let obj = {
+                    Service: dataArray[i].Service,
+                    IncidentType: dataArray[i].IncidentType,
+                    Category: dataArray[i].Category
+                };
+                let contains = false;
+                if (moduleName == 'categoriesIncidentTypes') {
+                    contains = await containsObjectCategories(obj, types);
+                } else {
+                    contains = await containsObjectServices(obj, types);
+                }
+
+                if (!contains) {
+                    obj.id = j;
+                    types.push(obj);
+                    j++;
+                }
+            }
+        }
+
+        if (moduleName == 'servicesIncidentTypes' || moduleName == 'categoriesIncidentTypes') {
+            dataArray = types;
+        }
+
+        let json = JSON.stringify(dataArray, null, '\t');
+        dataStore.save(moduleName, json);
+        if (data.recordCount < data.pageSize) {
+            moreRecords = false;
+        } else {
+            moreRecords = true;
+        }
+    } else {
+        moreRecords = false;
```

shoul throw an error in case of invalid response instead of returning `false`

---

In this case Cherwell api also can return 400 when no more records are found, Throw error on failed request on all of the others requests